src/ring_theory/ideal/operations.lean

```diff
@@ -2401,4 +2401,110 @@ lemma quot_quot_equiv_comm_algebra_map (x : R) :
+
+variables (I J : ideal A)
+
+/-- The natural algebra homomorphism `A / I → A / (I ⊔ J)`. -/
+def quot_left_to_quot_supₐ : A ⧸ I →ₐ[R] A ⧸ (I ⊔ J) :=
+alg_hom.mk (quot_left_to_quot_sup I J) rfl (map_mul _) rfl (map_add _) (λ _, rfl)
+
+/-- The kernel of `quot_left_to_quot_sup` -/
+@[simp]
+lemma ker_quot_left_to_quot_supₐ :
+  ring_hom.ker (quot_left_to_quot_supₐ R A I J) = J.map (I^.quotient.mkₐ R) :=
+ker_quot_left_to_quot_sup I J
+
+/-- The algebra homomorphism `(A / I) / J' -> A / (I ⊔ J) induced by `quot_left_to_quot_sup`,
+  where `J'` is the projection of `J` in `A / I`. -/
+def quot_quot_to_quot_supₐ : (A ⧸ I) ⧸ J.map (I^.quotient.mkₐ R) →ₐ[R] A ⧸ I ⊔ J :=
+alg_hom.mk (quot_quot_to_quot_sup I J) rfl (map_mul _) rfl (map_add _) (λ _, rfl)
+
+/-- The composition of the algebra homomorphisms `A → (A / I)` and `(A / I) → (A / I) / J'`,
+  where `J'` is the projection `J` in `A / I`. -/
+def quot_quot_mkₐ : A →ₐ[R] ((A ⧸ I) ⧸ J.map (I^.quotient.mkₐ R)) :=
+alg_hom.mk (quot_quot_mk I J) rfl (map_mul _) rfl (map_add _) (λ _, rfl)
+
+/-- The kernel of `quot_quot_mk`. -/
+@[simp]
+lemma ker_quot_quot_mkₐ : ring_hom.ker (quot_quot_mkₐ R A I J) = I ⊔ J :=
+ker_quot_quot_mk I J
+
+/-- The injective algebra homomorphism `A / (I ⊔ J) → (A / I) / J'`induced by `quot_quot_mk`,
+  where `J'` is the projection `J` in `A / I`. -/
+def lift_sup_quot_quot_mkₐ (I J : ideal A) :
+  A ⧸ (I ⊔ J) →ₐ[R] (A ⧸ I) ⧸ J.map (I^.quotient.mkₐ R) :=
+alg_hom.mk (lift_sup_quot_quot_mk I J) rfl (map_mul _) rfl (map_add _) (λ _, rfl)
+
+/-- `quot_quot_to_quot_add` and `lift_sup_quot_quot_mk` are inverse isomorphisms. In the case where
+    `I ≤ J`, this is the Third Isomorphism Theorem (see `quot_quot_equiv_quot_of_le`). -/
+def quot_quot_equiv_quot_supₐ : ((A ⧸ I) ⧸ J.map (I^.quotient.mkₐ R)) ≃ₐ[R] A ⧸ I ⊔ J :=
+@alg_equiv.of_ring_equiv R _ _ _ _ _ _ _ (quot_quot_equiv_quot_sup I J) (λ _, rfl)
+
+@[simp]
+lemma quot_quot_equiv_quot_sup_quot_quot_mkₐ (x : A) :
+  quot_quot_equiv_quot_supₐ R A I J (quot_quot_mkₐ R A I J x) = (I ⊔ J)^.quotient.mkₐ R x :=
+rfl
+
+@[simp]
+lemma quot_quot_equiv_quot_sup_symm_quot_quot_mkₐ (x : A) :
+  (quot_quot_equiv_quot_supₐ R A I J).symm ((I ⊔ J)^.quotient.mkₐ R x) =
+    quot_quot_mkₐ R A I J x :=
+rfl
+
+/-- The natural algebra isomorphism `(A / I) / J' → (A / J) / I'`,
+  where `J'` (resp. `I'`) is the projection of `J` in `A / I` (resp. `I` in `A / J`). -/
+def quot_quot_equiv_commₐ :
+  ((A ⧸ I) ⧸ J.map (I^.quotient.mkₐ R)) ≃ₐ[R]
+    ((A ⧸ J) ⧸ I.map (J^.quotient.mkₐ R)) :=
+@alg_equiv.of_ring_equiv R _ _ _ _ _ _ _ (quot_quot_equiv_comm I J) (λ _, rfl)
+
+@[simp]
+lemma quot_quot_equiv_comm_quot_quot_mkₐ (x : A) :
+  quot_quot_equiv_commₐ R A I J (quot_quot_mkₐ R A I J x) = quot_quot_mkₐ R A J I x :=
+rfl
+
+@[simp]
+lemma quot_quot_equiv_comm_comp_quot_quot_mkₐ :
+  alg_hom.comp ↑(quot_quot_equiv_commₐ R A I J) (quot_quot_mkₐ R A I J) = quot_quot_mkₐ R A J I :=
+alg_hom.ext $ quot_quot_equiv_comm_quot_quot_mkₐ R A I J
```

Comment:
I think the only lemmas worth keeping are the ones about `alg_hom.comp`, like this one. The rest are redundant if you add the lemmas I suggest above.